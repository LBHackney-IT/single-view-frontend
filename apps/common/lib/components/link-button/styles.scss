@import "~lbh-frontend/lbh/base";
@import "~lbh-frontend/lbh/core/links";
@import "~lbh-frontend/lbh/components/lbh-back-link/back-link";
