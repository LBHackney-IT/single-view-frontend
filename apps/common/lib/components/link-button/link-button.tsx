import React, { ComponentPropsWithoutRef, forwardRef } from "react";

import cn from "classnames";

import "./styles.scss";

export interface LinkButtonProps extends ComponentPropsWithoutRef<"button"> {
  variant?: "link" | "danger" | "text-colour" | "muted" | "back-link" | "native";
}

export const LinkButton = forwardRef<HTMLButtonElement, LinkButtonProps>(
  function LinkButton({ variant = "link", className, children, ...props }, ref) {
    const linkClasses = cn(
      variant !== "native" && {
        "govuk-link lbh-link": variant !== "back-link",
        "govuk-back-link lbh-back-link": variant === "back-link",
        [`lbh-link--${variant}`]: variant !== "link" && variant !== "back-link",
      },
      className,
    );
    return (
      <button ref={ref} className={linkClasses} type="button" {...props}>
        {children}
      </button>
    );
  },
);
