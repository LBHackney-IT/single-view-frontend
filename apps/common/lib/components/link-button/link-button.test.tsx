import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { LinkButton } from "./link-button";

test("it renders correctly", async () => {
  const { container } = render(<LinkButton>Test Button</LinkButton>);
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders a variant", () => {
  const { container } = render(<LinkButton variant="muted">Test Button</LinkButton>);
  expect(container).toMatchSnapshot();
});

test("it renders a back link", () => {
  const { container } = render(<LinkButton variant="back-link">Test Button</LinkButton>);
  expect(container).toMatchSnapshot();
});
