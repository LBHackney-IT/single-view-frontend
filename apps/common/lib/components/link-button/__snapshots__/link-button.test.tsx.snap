// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a back link 1`] = `
<div>
  <button
    class="govuk-back-link lbh-back-link"
    type="button"
  >
    Test Button
  </button>
</div>
`;

exports[`it renders a variant 1`] = `
<div>
  <button
    class="govuk-link lbh-link lbh-link--muted"
    type="button"
  >
    Test Button
  </button>
</div>
`;

exports[`it renders correctly 1`] = `
<div>
  <button
    class="govuk-link lbh-link"
    type="button"
  >
    Test Button
  </button>
</div>
`;
